pkg/challenges/challenges.go

```diff
@@ -69,6 +70,78 @@ type ChallengeResult struct {
 	AdditionalInfo map[AdditionalInfo]string
 }
```

Comment:
nit: Worth adding a comment that says this implements `identity.Principal`?

Reply:
Pushing a recent change that makes that a lot more clear I think

Reply:
Yep, I think this looks clear!